library/src/animationManager.cxx

```diff
@@ -274,3 +274,37 @@ bool animationManager::LoadAtTime(double timeValue)
   return true;
 }
 }
+void f3d::detail::animationManager::CycleAnimation()
+{
+
+  assert(this->Importer);
+  int nIndex = this->Importer->GetNumberOfAnimations();
+  if (nIndex <= 0)
+  {
+    return;
+  }
+  if (this->HasAnimation)
+  {
+    this->AnimationIndex = (this->AnimationIndex + 1) % nIndex;
```

Comment:
If I remember well, -1 means "All animations enabled", so you basically want a loop from `-1` to `N-1`.

Reply:
Correct me if I am wrong. I left it with a loop from `0` to `N-1`. The animationManager initialize function sets it at `-1` from the beginning. From what I understand, `-1` is a sentinel value used only for the logic within the initialize function. When I include `-1` within the CycleAnimation() loop, then there are no animations to play at `-1` which feels awkward. Therefore, I removed that value from the loop.

Reply:
-1 is all animations :)

Reply:
Conversation moved to F3D Discord. `./bin/f3d ../src/testing/data/InterpolationTest.glb` was shared as a good example for understanding "-1 = all animations".